cmd/kubectl-gadget/deploy.go

```diff
@@ -236,30 +240,43 @@ func init() {
 	// WARNING For now, use StringVar() instead of StringSliceVar() as only the
 	// first line of the file will be taken when used with
 	// --gadgets-public-keys="$(cat inspektor-gadget.pub),$(cat your-key.pub)"
 	deployCmd.PersistentFlags().StringVar(
 		&gadgetsPublicKeys,
 		"gadgets-public-keys", resources.InspektorGadgetPublicKey, "Public keys used to verify the gadgets")
+	deployCmd.PersistentFlags().MarkDeprecated("gadgets-public-keys", "This flag is deprecated and will be removed in a future release. Use --configmap-override-path instead")
 	deployCmd.PersistentFlags().StringSliceVar(
 		&allowedGadgets,
 		"allowed-gadgets", []string{}, "List of allowed gadgets, if gadget is not part of it, execution will be denied. By default, all gadgets are allowed.")
+	deployCmd.PersistentFlags().MarkDeprecated("allowed-gadgets", "This flag is deprecated and will be removed in a future release. Use --override-configmap instead")
 	deployCmd.PersistentFlags().StringSliceVar(
 		&insecureRegistries,
 		"insecure-registries",
 		[]string{},
 		"List of registries to access over plain HTTP",
 	)
+	deployCmd.PersistentFlags().MarkDeprecated("insecure-registries", "This flag is deprecated and will be removed in a future release. Use --configmap-override-path instead")
 	deployCmd.PersistentFlags().BoolVar(
 		&disallowGadgetsPull,
 		"disallow-gadgets-pulling", false, "Disallow pulling gadgets from registries")
+	deployCmd.PersistentFlags().MarkDeprecated("disallow-gadgets-pulling", "This flag is deprecated and will be removed in a future release. Use --set-configmap=operator.oci.disallow-pulling=<true|false> instead")
 	deployCmd.PersistentFlags().BoolVar(
 		&otelMetricsListen,
 		"otel-metrics-listen", false, "Enable OpenTelemetry metrics listener (Prometheus compatible) endpoint")
+	deployCmd.PersistentFlags().MarkDeprecated("otel-metrics-listen", "This flag is deprecated and will be removed in a future release. Use --set-configmap=operator.otel-metrics.otel-metrics-listen=<true|false> instead")
 	deployCmd.PersistentFlags().StringVar(
 		&otelMetricsListenAddr,
 		"otel-metrics-listen-address", "0.0.0.0:2224", "Address and port to create the OpenTelemetry metrics listener (Prometheus compatible) on")
+	deployCmd.PersistentFlags().MarkDeprecated("otel-metrics-listen-address", "This flag is deprecated and will be removed in a future release. Use --set-configmap=operator.otel-metrics.otel-metrics-listen-address=<address> instead")
+	deployCmd.PersistentFlags().StringSliceVar(
+		&setConfigmap,
+		"set-configmap", []string{}, "Set a key-value pair in the ConfigMap. The format is key=value. This flag can be used multiple times to set multiple values. Only used for primitive values (string, int, bool). For complex values, use --configmap-override-path.")
+	deployCmd.PersistentFlags().StringVar(
+		&configmapOverridePath,
+		"configmap-override-path", "", "Path to a config file to override the values in the ConfigMap. The file should be in YAML format")
```

Comment:
I'm not sure if people care about this configs being stored into a configmap. I will investigate how other projects do it, but IMO, this could simply be `config` as it's for `ig` and not mention configmap at all.

Reply:
I want to be clear here so people don't confuse this with https://inspektor-gadget.io/docs/v0.41.0/reference/configuration#configuration-file. Probably we can come up with a better name but using configmap for your Daemonset is well-know and essentially the goal of this change is to allow user easily configure this configmap so hence this name!

Reply:
Wait, what values is that config file configuring? I didn't remember it 😅